tests/subprocesstest.py

```diff
@@ -0,0 +1,125 @@
+"""
+subprocesstest enables unittest test cases and suites to be run in separate
+python interpreter instances, in parallel.
+
+A base class SubprocessTestCase is provided that, when extended, will run test
+cases marked with @run_in_subprocess in a separate python interpreter.
+"""
+import unittest
+import subprocess
+import sys
+
+
+SUBPROC_TEST_ATTR = '_subproc_test'
+
+
+def run_in_subprocess(obj):
+    """
+    Marks a test case that is to be run in its own 'clean' interpreter instance.
+
+    When applied to a TestCase class, each method will be run in a separate
+    interpreter instance, in parallel.
+
+    Usage on a class::
+
+        @run_in_subprocess
+        class PatchTests(object):
+            # will be run in new interpreter
+            def test_patch_before_import(self):
+                patch()
+                import module
+
+            # will be run in new interpreter as well
+            def test_patch_after_import(self):
+                import module
+                patch()
+
+
+    Usage on a test method::
+
+        class OtherTests(object):
+            @run_in_subprocess
+            def test_case(self):
+                pass
+
+
+    :param obj: method or class to run in a separate python interpreter.
+    :return:
+    """
+    setattr(obj, SUBPROC_TEST_ATTR, True)
```

Comment:
I'd love to future proof this helper a bit.

I can see a future where we pass additional data to this helper function, e.g. environment variables ?

```python
def run_in_subprocess(func_or_options):
    if inspect.isfunction(func_or_options) or inspect.isclass(func_or_options):
        setattr(func_or_options, SUBPROC_TEST_ATTR, True)
        return func_or_options

    options = func_or_options
    def wrapper(obj):
        setattr(obj, SUBPROC_TEST_ATTR, True)
        setattr(obj, SUBPROC_ENV_ATTR, options.get('env'))
        return obj
    return wrapper
```


Then can use as:

```python
@run_in_subprocess(dict(env=dict(DD_TRACE_DEBUG='true')))
def test_dd_trace_debug(self):
    assert os.environ.get('DD_TRACE_DEBUG', 'true')
```


There is a good chance I am complicating things here... and we can always come back and visit this in the future if we ever need this functionality, just thinking of possible futures this wrapper could have.

Reply:
Great ideas. But I think this is outside the scope of this PR.

Reply:
agreed